src/Sulu/Bundle/WebsiteBundle/Resources/config/website.xml

```diff
@@ -5,13 +5,39 @@
 
     <parameters>
         <parameter key="sulu_website.data_collector.sulu_collector.class">Sulu\Bundle\WebsiteBundle\DataCollector\SuluCollector</parameter>
+        <parameter key="sulu_website.engine.initialize" type="constant">Sulu\Bundle\WebsiteBundle\Templating\EngineEvents::INITIALIZE</parameter>
     </parameters>
 
     <services>
         <!-- request analyzer data collector -->
         <service id="sulu_website.data_collector.sulu_collector" class="%sulu_website.data_collector.sulu_collector.class%">
             <argument type="service" id="sulu_core.webspace.request_analyzer" />
+
             <tag name="data_collector" template="SuluWebsiteBundle:Profiler:layout" id="sulu"/>
+            <tag name="sulu.context" context="website"/>
+        </service>
+
+        <service id="sulu_website.provider.content" class="Sulu\Bundle\WebsiteBundle\Routing\ContentRouteProvider">
+            <argument type="service" id="sulu_core.webspace.request_analyzer"/>
+
+            <tag name="sulu.context" context="website"/>
+        </service>
+
+        <service id="sulu_website.templating.eventable_engine"
+                 class="Sulu\Bundle\WebsiteBundle\Templating\EventableEngine" decorates="templating">
```

Comment:
Didn't I hear today morning that you want to look out for a better name? What about `EventAwareEngine`, corresponding to the `ContainerAware...` classes?